### PR TITLE
cumulus test runtime: remove `GenesisExt`

### PR DESCRIPTION
**Pull request by michalkucharczyk**
This PR removes the `GenesisExt` wrapper over the `GenesisRuntimeConfig` in `cumulus-test-service`. Initialization of values that were performed by `GenesisExt::BuildStorage` was moved into  `test_pallet` genesis.

### CODE REVIEW

File: cumulus/test/runtime/src/lib.rs [hunk 1]
```diff
@@ -367,7 +364,10 @@ pub struct TestOnRuntimeUpgrade;
 
 impl OnRuntimeUpgrade for TestOnRuntimeUpgrade {
 	fn on_runtime_upgrade() -> frame_support::weights::Weight {
-		assert_eq!(sp_io::storage::get(TEST_RUNTIME_UPGRADE_KEY), Some(vec![1, 2, 3, 4].into()));
+		assert_eq!(
+			sp_io::storage::get(test_pallet::TEST_RUNTIME_UPGRADE_KEY),
+			Some(vec![1, 2, 3, 4].into())
+		);
```
Comment (michalkucharczyk) [inline cumulus/test/runtime/src/lib.rs:370]: @bkchr do we still need it?

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4200860

**Comment by michalkucharczyk** [comment]
bot clean

File: cumulus/test/runtime/src/lib.rs (hunk 1 above)
Comment (bkchr) [reply cumulus/test/runtime/src/lib.rs:370]: I assume there was a bug. I have gone back in history to find out why I introduced this, but I didn't find anything :see_no_evil:

File: cumulus/test/runtime/src/test_pallet.rs [hunk 2]
```diff
@@ -17,8 +17,15 @@
 /// A special pallet that exposes dispatchables that are only useful for testing.
 pub use pallet::*;
 
+/// Some key that we set in genesis and only read in [`TestOnRuntimeUpgrade`] to ensure that
+/// [`OnRuntimeUpgrade`] works as expected.
+pub const TEST_RUNTIME_UPGRADE_KEY: &[u8] = b"+test_runtime_upgrade_key+";
+/// Allows to set `runtime::ParachainId` in genesis:
+pub const PARACHAIN_ID_RUNTIME_KEY: &[u8] = b":ParachainId:";
```
Comment (bkchr) [inline cumulus/test/runtime/src/test_pallet.rs:24]:
```suggestion
```

File: cumulus/test/runtime/src/test_pallet.rs [hunk 3]
```diff
@@ -45,4 +52,25 @@ pub mod pallet {
...
+
+	#[pallet::genesis_build]
+	impl<T: Config> BuildGenesisConfig for GenesisConfig<T> {
+		fn build(&self) {
+			sp_io::storage::set(TEST_RUNTIME_UPGRADE_KEY, &[1, 2, 3, 4].encode());
+			self.self_para_id.map(|para_id| {
+				sp_io::storage::set(
+					&sp_io::hashing::twox_128(&PARACHAIN_ID_RUNTIME_KEY),
+					&para_id.encode(),
+				)
```
Comment (bkchr) [inline cumulus/test/runtime/src/test_pallet.rs:72]:
```suggestion
				super::ParachainId::set(para_id);
```

**Review by bkchr — APPROVED** [review]

File: cumulus/test/runtime/src/lib.rs (hunk 1 above)
Comment (michalkucharczyk) [reply cumulus/test/runtime/src/lib.rs:370]: I'd remove it, as it makes little sense. Has nothing to do with runtime upgrade.

Would make sense if upgrade would wipe the storage.

Comment (bkchr) [reply cumulus/test/runtime/src/lib.rs:370]: Let's remove it. IDK. I should have left better comments...

File: cumulus/test/service/src/chain_spec.rs [hunk 4]
```diff
@@ -86,16 +66,12 @@ pub fn get_chain_spec_with_extra_endowed(
 	id: ParaId,
 	extra_endowed_accounts: Vec<AccountId>,
 ) -> ChainSpec {
+	#[allow(deprecated)]
```
Comment (skunert) [inline cumulus/test/service/src/chain_spec.rs:69]: What is deprecated here?

**Review by skunert — APPROVED** [review]

**Review by skunert — APPROVED** [review]

File: cumulus/test/service/src/chain_spec.rs (hunk 4 above)
Comment (michalkucharczyk) [reply cumulus/test/service/src/chain_spec.rs:69]: I was already in alternative future.

File: cumulus/test/runtime/src/lib.rs (hunk 1 above)
Comment (michalkucharczyk) [reply cumulus/test/runtime/src/lib.rs:370]: OK, you convinced me, I have no idea. It won't hurt us. And the assert is correct. Let it be here.

File: cumulus/test/service/src/chain_spec.rs (hunk 4 above)
Comment (michalkucharczyk) [inline cumulus/test/service/src/chain_spec.rs:69]:
```suggestion
```

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4209314

File: cumulus/test/runtime/src/test_pallet.rs [hunk 5]
```diff
@@ -45,4 +50,22 @@ pub mod pallet {
...
+		#[serde(skip)]
+		pub _config: sp_std::marker::PhantomData<T>,
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config> BuildGenesisConfig for GenesisConfig<T> {
+		fn build(&self) {
+			sp_io::storage::set(TEST_RUNTIME_UPGRADE_KEY, &[1, 2, 3, 4]);
+			self.self_para_id.map(|para_id| {
+				super::ParachainId::set(para_id);
```
Comment (michalkucharczyk) [inline cumulus/test/runtime/src/test_pallet.rs:67]:
```suggestion
				crate::ParachainId::set(&para_id);
```

File: cumulus/test/runtime/src/test_pallet.rs [hunk 6]
```diff
@@ -17,8 +17,13 @@
 /// A special pallet that exposes dispatchables that are only useful for testing.
 pub use pallet::*;
 
+/// Some key that we set in genesis and only read in [`TestOnRuntimeUpgrade`] to ensure that
+/// [`OnRuntimeUpgrade`] works as expected.
+pub const TEST_RUNTIME_UPGRADE_KEY: &[u8] = b"+test_runtime_upgrade_key+";
+
 #[frame_support::pallet(dev_mode)]
 pub mod pallet {
+	use crate::test_pallet::{PARACHAIN_ID_RUNTIME_KEY, TEST_RUNTIME_UPGRADE_KEY};
```
Comment (michalkucharczyk) [inline cumulus/test/runtime/src/test_pallet.rs:26]:
```suggestion
	use crate::test_pallet::TEST_RUNTIME_UPGRADE_KEY;
```

**Pull request merged**